CMakeLists.txt

```diff
@@ -206,8 +206,10 @@ if(ENABLE_ISHMEM)
       NAME ${test_name}
       COMMAND
         ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${processes}
-        ${MPIEXEC_PREFLAGS} ${CMAKE_BINARY_DIR}/bin/ishmrun ./${name} ${ARGN}
-        COMMAND_EXPAND_LISTS)
+        ${MPIEXEC_PREFLAGS} -outfile-pattern "${test_name}.out.%r.log"
+        -errfile-pattern "${test_name}.err.%r.log"
+        ${CMAKE_BINARY_DIR}/bin/ishmrun gdb --ex r -ex bt --ex q --batch --args
```

Comment:
this version of `add_mhp_ctest` is used for all mhp tests if compiling with ishmem is enabled

`ishmrun` does the same which mhp::select_queue does (except round robin) by setting ZE_AFFINITY before running binary. ISHMEM requires it, DR handles it correctly thanks to round robin

I'd like to propose adding running tests under `gdb` with `bt` command after the test because in case of failure/segfault one can see backtraces right in the log files. What do you think? Do you have any idea of getting backtraces in nicer/cleaner/shorter way? (e.g. run gdb just in case of failure, but how to make CI dump corefiles)

Reply:
I asked because it used gdb to run, but if it is only ishmem then it does not matter.

impi set affinity, but mpich does not, so I added the round robin. ishmem only needs to support one launcher.

Reply:
>  impi set affinity, but mpich does not, so I added the round robin. ishmem only needs to support one launcher.

Ishmem developers decided to provide wrapper (ishmrun) script which sets affinity. To have things clean I consider adding same script to dist-ranges, which will allow to remove round robin code, remove need for ishmrun script and our way of executing code will be unified for ishmem and MPI. But this is not urgent. If you have an opinion about this (add script, or not to add) please share. If not let's postpone discussion for later time.